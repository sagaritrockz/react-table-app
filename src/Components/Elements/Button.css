.btn {
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 1px solid transparent;
    padding: 5px 10px;
    font-size: 14px;
    line-height: 1.5;
    border-radius: 0.25rem;
    cursor: pointer;
    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
        border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
.btn-custom {
    margin: 0 5px;
}