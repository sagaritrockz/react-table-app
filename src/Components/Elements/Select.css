.select {
    display: inline-block;
    padding: 5px 10px;
    font-size: 14px;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    cursor: pointer;
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
    text-transform: capitalize;
}
.select-custom {
    padding: 5px 10px 8px;
    margin: 0 5px;
}
