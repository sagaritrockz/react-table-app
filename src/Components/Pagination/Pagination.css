.pagination-container {
	display: flex;
	justify-content: space-between;
}
.pagination-text {
	padding: 10px 10px;
	color: grey;
	font: 400 13.3333px Arial;
	font-style: italic;
}
.pagination {
	display: inline-block;
}
