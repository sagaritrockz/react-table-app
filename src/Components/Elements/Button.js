import React from "react"
import { string, func } from 'prop-types'
import "./Button.css"

const Button = ({ label, click, name }) => {
    return (
        <button className="btn btn-custom" onClick={click} name={name}>
            {label}
        </button>
    );
};

Button.defaultProps = {
    label: "",
    click: f => f,
    name: ""
};

Button.propTypes = {
    label: string.isRequired,
    click: func.isRequired,
    name: string.isRequired
};

export default Button;
