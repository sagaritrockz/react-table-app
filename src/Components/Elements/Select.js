import React from "react"
import "./Select.css"

const Select = ({ list, change, value, name, show }) => {

    const onChange = e => {
        e.preventDefault();
        change(e.target.name, e.target.value);
    };
    
    if (show) {
        return (
            <select
                className="select select-custom"
                onChange={onChange}
                name={name}
                value={value}
            >
                <option value={0}>Select</option>
                {list.map((item, index) => (
                    <option key={index} value={item.id ? item.id : index + 1} >
                        {item.value ? item.value : item}
                    </option>
                ))}
            </select>
        );
    } else {
        return (
            <select
                className="select select-custom"
                onChange={onChange}
                name={name}
                value={value}
            >
                {list.map((item, index) => (
                    <option key={index} value={item.id ? item.id : index + 1} >
                        {item.value ? item.value : item}
                    </option>
                ))}
            </select>
        );
    }
};

Select.defaultProps = {
    list: [],
    change: f => f, 
    value: '', 
    name: '',
    show: false
}

export default Select;
