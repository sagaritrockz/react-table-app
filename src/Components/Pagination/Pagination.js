import React from "react";
import { number, string, func } from "prop-types";
import Select from "../Elements/Select";
import Button from "../Elements/Button";
import "./Pagination.css";

const Pagination = ({
	total,
	perPage,
	page,
	pages,
	pageKey,
	paginate,
	updateState,
	dropdownKey,
	dropdownVal
}) => {
	let to = perPage * page;
	let from = to - perPage + 1;
	const onNext = e => {
		e.preventDefault();
		if (page + 1 <= pages) {
			paginate(pageKey, page + 1);
		}
	};
	const onPrev = e => {
		e.preventDefault();
		if (page - 1 >= 1) {
			paginate(pageKey, page - 1);
		}
	};
	const onSelect = (name, value) => {
		paginate(pageKey, parseInt(value, 10));
		updateState(name, value);
	};

	return (
		<div className="pagination-container">
			<div class="pagination-text">
				Showing {from} to {to < total ? to : total} of {total} records
			</div>
			<div className="pagination">
				<Button label="Prev" name={pageKey} click={onPrev} />
				<Select
					list={Array.from(
						new Array(pages),
						(val, index) => index + 1
					)}
					value={dropdownVal}
					change={onSelect}
					name={dropdownKey}
				/>
				<Button label="Next" name={pageKey} click={onNext} />
			</div>
		</div>
	);
};

Pagination.defaultProps = {
	total: 10,
	perPage: 10,
	page: 1,
	pages: 1,
	pageKey: "",
	paginate: f => f,
	updateState: f => f,
	dropdownKey: "",
	dropdownVal: ""
};

Pagination.propTypes = {
	total: number.isRequired,
	perPage: number.isRequired,
	page: number.isRequired,
	pages: number.isRequired,
	pageKey: string.isRequired,
	paginate: func.isRequired,
	updateState: func.isRequired,
	dropdownKey: string.isRequired,
	dropdownVal: string.isRequired
};

export default Pagination;
