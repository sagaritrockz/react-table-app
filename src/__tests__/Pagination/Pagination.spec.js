import React from 'react'
import Enzyme from 'enzyme'
import Adaptar from 'enzyme-adapter-react-16.3'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Pagination from '../../Components/Pagination/Pagination'
import Select from "../../Components/Elements/Select"
import Button from "../../Components/Elements/Button"

Enzyme.configure({adapter: new Adaptar()});

describe('Pagination', () => {
    it('should render correctly', () => {
        const 
            total = 304,
            perPage = 10,
            page = 1,
            pages = 31,
            paginate = function() {},
            pageKey = "_page",
            dropdownKey = "_paginationDropdown",
            dropdownVal = "_page",
            updateState = function(){};

        const output = shallow(
            <Pagination
                total={total}
                perPage={parseInt(perPage, 10)}
                page={page}
                pages={pages}
                paginate={paginate}
                pageKey={pageKey}
                dropdownKey={dropdownKey}
                dropdownVal={dropdownVal}
                updateState={updateState}
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});