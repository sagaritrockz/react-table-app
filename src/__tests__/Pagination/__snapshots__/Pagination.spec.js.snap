// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<div
  className="pagination-container"
>
  <div
    className="pagination"
  >
    <Button
      click={[Function]}
      label="Prev"
      name="_page"
    />
    <Select
      change={[Function]}
      defaultOption=""
      list={
        Array [
          1,
          2,
          3,
          4,
          5,
          6,
          7,
          8,
          9,
          10,
          11,
          12,
          13,
          14,
          15,
          16,
          17,
          18,
          19,
          20,
          21,
          22,
          23,
          24,
          25,
          26,
          27,
          28,
          29,
          30,
          31,
        ]
      }
      name="_paginationDropdown"
      value="_page"
    />
    <Button
      click={[Function]}
      label="Next"
      name="_page"
    />
  </div>
</div>
`;
